File: 151002/Shydlouski/Lab1/publisher/src/main/java/org/example/publisher/impl/tag/mapper/TagMapper.java
package org.example.publisher.impl.tag.mapper;
import org.example.publisher.impl.tag.Tag;
import org.example.publisher.impl.tag.dto.TagRequestTo;
import org.example.publisher.impl.tag.dto.TagResponseTo;
import org.example.publisher.impl.tweet.Tweet;

import java.util.List;

public interface TagMapper {
    TagRequestTo tagToRequestTo(Tag tag);

    List<TagRequestTo> tagToRequestTo(Iterable<Tag> tags);

    Tag dtoToEntity(TagRequestTo tagRequestTo, List<Tweet> tweets);

    TagResponseTo tagToResponseTo(Tag tag);

    List<TagResponseTo> tagToResponseTo(Iterable<Tag> tags);
}

File: 151003/Alekseeva/TaskNoSQL/Module.Publisher/Mappers/TagMapper.cs
using Publisher.Dto.Request.CreateTo;
using Publisher.Dto.Request.UpdateTo;
using Publisher.Dto.Response;
using Publisher.Models;
using Riok.Mapperly.Abstractions;
namespace Publisher.Mappers;

[Mapper]
public static partial class TagMapper
{
    public static partial Tag ToEntity(this UpdateTagRequestTo updateTagRequestTo);
    public static partial Tag ToEntity(this CreateTagRequestTo createTagRequestTo);
    public static partial TagResponseTo ToResponse(this Tag tag);
    public static partial IEnumerable<TagResponseTo> ToResponse(this IEnumerable<Tag> tags);
}